libs/datamodel/core/src/transform/ast_to_dml/db/attributes.rs

```diff
@@ -158,9 +158,14 @@ pub(super) fn validate_relation_fields(ctx: &mut Context<'_>) {
         let model = ctx.db.walk_model(*model_id);
         let field = model.walk_relation_field(*field_id);
 
+        relation::validate_same_length_in_referencing_and_referenced(field, &mut errors);
         relation::validate_relation_field_arity(field, &mut errors);
         relation::validate_ignored_related_model(field, &mut errors);
         relation::validate_on_update_without_foreign_keys(field, referential_integrity, &mut errors);
+
+        // we don't want to spam errors, so run this the last. order matters.
+        relation::validate_references_unique_fields(field, ctx.db.active_connector(), &mut errors);
```

Comment:
Also, nitpick on this line, but is it really a relation attribute validation? We can think about changing the module structure (doesn't have to happen in this PR).

Reply:
These are only run if there are no other errors. Otherwise many of the errors we can trigger triggers also these validations. Therefore there's a check we do not have any errors before we run these, and we have tests that do fail if the order is not correct.

Reply:
> Also, nitpick on this line, but is it really a relation attribute validation? We can think about changing the module structure (doesn't have to happen in this PR).

Isn't `references` a relation attribute?

Reply:
Yes, but it can be left implicit. What is taken into account in the walkers as they are now are only the explicit ones.

Reply:
After discussing on Slack, I agree we should do many of these validations with a different set of walkers. Some of the validations validate just what attributes the fields have, and they can be done here. The other functions operate on relations, and we should have a separate round for them, where we validate fully constructed relations.

I'm ok with both: merging or not merging this. If we merge, we must rewrite some of the validations, but it would be much easier due to this code here being a bit more close to whatever the relations interface would be, so it might make sense to merge. WDYT @tomhoule

Reply:
I think we can merge